__DEFINES/_macros.dm

```diff
@@ -32,7 +32,7 @@
 
 #define isgolem(A) (ishuman(A) && A.species && istype(A.species, /datum/species/golem))
 
-#define isslimeperson(A) (ishuman(A) && A.species && istype(A.species, /datum/species/slime))
+#define isslimeperson(A) (ishuman(A) && istype(A:species, /datum/species/slime))
```

Comment:
All these macros are dumb, they assume your `A` is already of type human, because they try to read the `species` var, but they also check for `ishuman(A)`.
They should all be changed to something like this.